pkg/crypto/hash_test.go

```diff
@@ -0,0 +1,66 @@
+package crypto
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHash(t *testing.T) {
+	cases := []struct {
+		name           string
+		input          []interface{}
+		expectedOutput string
+		expectedError  error
+	}{
+		{
+			name:           "hash a string",
+			input:          []interface{}{"foobar"},
+			expectedOutput: "09234807e4af85f17c66b48ee3bca89dffd1f1233659f9f940a2b17b0b8c6bc5",
```

Comment:
Is this correct for our intial use case? to create he index names? are you planning on just using this string with now other prefix? it is already 64 characters, so any prefix will cause it to truncate

Reply:
In the initial usecase I would just use the first 16 or 32 bytes of this string. The collision resistance will still be high enough.

Reply:
is there a good reason to not just add a "ShortHasher" implementation that does this and is documented? i suspect this usecase of "i need these values hashed and only X characters output" is going to keep happening

Reply:
I think that would be more work than it would save time, even in the long run. What we would save is that we could write something like this:
```go
hash, err := crypto.HashN(16, "foobar")
```
instead of 
```go
hash, err := crypto.Hash(16, "foobar")
...
hash = hash[:16]
```

Reply:
fair enough